### PR TITLE
relax the requirement of onnxruntime to >=1.16.3

### LINKED ISSUE #32: The package seems to work with onnxruntime==1.6.3, suggesting relaxing the requirement.
**Is your feature request related to a problem? Please describe.**
onnxruntime dropped support for CentOS [starting 1.17.0](https://github.com/microsoft/onnxruntime/releases/tag/v1.17.0), and unfortunately lots of scientific clusters are still on CentOS, including the one I am using. This means pip refuses to install anything later than 1.16.3 and resulting in installation fails.

**Describe the solution you'd like**
Relax the requirements to >=1.16.3, with some deeper test.

**Additional context**
I have tested the simple demo on readme page, and it worked as expected with onnxruntime==1.6.3.
Here is a full list of all the packages I have. (everything except onnxruntime satisfy the requirements.txt)

Package                  Version
------------------------ -----------
aiohappyeyeballs         2.6.1
aiohttp                  3.11.18
aiosignal                1.3.2
asttokens                3.0.0
async-timeout            5.0.1
attrs                    25.3.0
biopython                1.85
CAI-PyPI                 2.0.1
certifi                  2025.4.26
charset-normalizer       3.4.2
click                    8.1.8
coloredlogs              15.0.1
comm                     0.2.2
decorator                5.2.1
exceptiongroup           1.3.0
executing                2.2.0
filelock                 3.18.0
flatbuffers              25.2.10
frozenlist               1.6.0
fsspec                   2025.5.0
huggingface-hub          0.31.4
humanfriendly            10.0
idna                     3.10
ipython                  8.18.1
ipywidgets               8.1.7
jedi                     0.19.2
jinja2                   3.1.6
joblib                   1.5.0
jupyterlab-widgets       3.0.15
lightning-utilities      0.14.3
MarkupSafe               3.0.2
matplotlib-inline        0.1.7
mpmath                   1.3.0
multidict                6.4.4
networkx                 3.2.1
numpy                    1.26.4
nvidia-cublas-cu12       12.4.5.8
nvidia-cuda-cupti-cu12   12.4.127
nvidia-cuda-nvrtc-cu12   12.4.127
nvidia-cuda-runtime-cu12 12.4.127
nvidia-cudnn-cu12        9.1.0.70
nvidia-cufft-cu12        11.2.1.3
nvidia-curand-cu12       10.3.5.147
nvidia-cusolver-cu12     11.6.1.9
nvidia-cusparse-cu12     12.3.1.170
nvidia-cusparselt-cu12   0.6.2
nvidia-nccl-cu12         2.21.5
nvidia-nvjitlink-cu12    12.4.127
nvidia-nvtx-cu12         12.4.127
onnxruntime              1.16.3
packaging                25.0
pandas                   2.2.3
parso                    0.8.4
pexpect                  4.9.0
pip                      20.2.3
prompt-toolkit           3.0.51
propcache                0.3.1
protobuf                 6.31.0
ptyprocess               0.7.0
pure-eval                0.2.3
pygments                 2.19.1
python-codon-tables      0.1.18
python-dateutil          2.9.0.post0
pytorch-lightning        2.5.1.post0
pytz                     2025.2
PyYAML                   6.0.2
regex                    2024.11.6
requests                 2.32.3
safetensors              0.5.3
scikit-learn             1.6.1
scipy                    1.13.1
setuptools               80.8.0
six                      1.17.0
stack-data               0.6.3
sympy                    1.14.0
threadpoolctl            3.6.0
tokenizers               0.21.1
torch                    2.6.0
torchmetrics             1.7.1
tqdm                     4.67.1
traitlets                5.14.3
transformers             4.52.1
triton                   3.2.0
typing-extensions        4.13.2
tzdata                   2025.2
urllib3                  2.4.0
wcwidth                  0.2.13
widgetsnbextension       4.0.14
yarl                     1.20.0

### PR DESCRIPTION
**Pull request by ym3141**
Should resolves #32

### CODE REVIEW

**Comment by Adibvafa** [comment]
Thank you for this PR! I will run the tests now.

**Review by Adibvafa — COMMENTED** [review]
LGTM
